app/api/teams.py

```diff
@@ -219,3 +221,219 @@ async def extend_team_trial(
+                return TeamMergeResponse(
+                    success=False,
+                    message=f"Merge cancelled due to {len(conflicts)} key name conflicts",
+                    conflicts_resolved=conflicts,
+                    keys_migrated=0,
+                    users_migrated=0
+                )
+
+            source_keys = await _resolve_key_conflicts(
+                conflicts,
+                merge_request.conflict_resolution_strategy,
+                source_keys,
+                merge_request.rename_suffix,
+                db,
+                current_user
+            )
+
+        # Store team names before deletion
+        source_team_name = source_team.name
+        target_team_name = target_team.name
+
+        # Migrate users and keys
+        users_migrated = await _migrate_users_to_team(source_users, target_team.id, db)
+        keys_migrated = await _migrate_keys_to_team(source_keys, target_team.id, db)
+
+        # Update LiteLLM key associations
+        # Create a map of keys by region to avoid unnecessary DB queries
+        keys_by_region = {}
+        for key in source_keys:
+            if key.region_id not in keys_by_region:
+                keys_by_region[key.region_id] = []
+            keys_by_region[key.region_id].append(key)
+
+        # Update LiteLLM key associations for each region
+        for region_id, region_keys in keys_by_region.items():
+            # Get region info
+            region = db.query(DBRegion).filter(
+                DBRegion.id == region_id,
+                DBRegion.is_active == True
+            ).first()
+
+            if not region:
+                continue
+
+            # Initialize LiteLLM service for this region
+            litellm_service = LiteLLMService(
+                api_url=region.litellm_api_url,
+                api_key=region.litellm_api_key
+            )
+
+            # Update team association for each key in this region
+            for key in region_keys:
+                try:
+                    await litellm_service.update_key_team_association(
+                        key.litellm_token,
+                        f"{region.name.replace(' ', '_')}_{target_team.id}"
+                    )
+                except Exception as e:
+                    logger.error(f"Failed to update LiteLLM key {key.id}: {str(e)}")
+                    # Continue with other keys even if one fails
+
+        # Delete source team
+        db.delete(source_team)
+        db.commit()
+
+        return TeamMergeResponse(
+            success=True,
+            message=f"Successfully merged team '{source_team_name}' into '{target_team_name}'",
+            conflicts_resolved=conflicts if conflicts else None,
+            keys_migrated=keys_migrated,
+            users_migrated=users_migrated
+        )
+
+    except HTTPException:
+        # Re-raise HTTP exceptions as-is
+        raise
+    except Exception as e:
+        db.rollback()
```

Comment:
will this actually work if commits have been made in the `_migrate_keys_to_team` and `_migrate_users_to_team` functions?